lib/argument.js

```diff
@@ -47,6 +50,24 @@ var argumentValidation = {
     if (!callback || !_.isFunction(callback)) {
       throw new Error('acquireToken requires a function callback parameter.');
     }
+  }, 
+
+  validateUserCodeInfo : function(userCodeInfo) {
+     if (!userCodeInfo){
+        throw new Error('The userCodeInfo parameter is required');
+     }
+
+     if (!userCodeInfo.hasOwnProperty(userCodeResponseFields.DEVICE_CODE)){
```

Comment:
userCodeResponseFields is an constant, and before checing if userCodeInfo has the property of device_code, we've already check to make sure it's non-null.